daemon/job_actual_ip.go

```diff
@@ -0,0 +1,67 @@
+package daemon
+
+import (
+	"context"
+	"log"
+	"net/netip"
+	"time"
+
+	"github.com/NordSecurity/nordvpn-linux/core"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/network"
+)
+
+func insightsUntilSuccess(ctx context.Context, api core.InsightsAPI) (core.Insights, error) {
+	for i := 0; ; i++ {
+		select {
+		case <-ctx.Done():
+			return core.Insights{}, ctx.Err()
+		default:
+			insights, err := api.Insights()
+			if err == nil && insights != nil {
+				return *insights, nil
+			} else {
+				log.Println(internal.ErrorPrefix, err)
+			}
+
+			backoff := network.ExponentialBackoff(i)
+
+			// Wait before retrying
+			select {
+			case <-time.After(backoff):
+				// Continue to the next retry
+			case <-ctx.Done():
+				return core.Insights{}, ctx.Err() // Exit if context is canceled during sleep
+			}
+		}
+	}
+
+}
+
+func JobActualIP(dm *DataManager, api core.InsightsAPI) func(context.Context, bool) error {
+	return func(ctx context.Context, isConnected bool) error {
+		var newIP netip.Addr
+		defer func() {
```

Comment:
The actual IP will be set to zero value if there was an error in the job, even if the IP was set to something else previously. Is this intentional? I guess it's better to not show the wrong IP if we don't have data on the new IP (or confirmation that the current IP is still correct), so maybe put a comment here that explains it, so it won't be confusing in the future